docs/api-spec.md

```diff
@@ -48,269 +72,447 @@ There is no formal specification of the Kubernetes API and Resource Model. This
 
-Its HTTP API endpoint is `/apis/tekton.dev/v1beta1/[parent]/taskruns`,  where  `[parent]` can refer to any string identifying a grouping of `TaskRun`s.
+| Field        | Type                              | Requirement | Notes                                             |
+| ------------ | --------------------------------- | ----------- | ------------------------------------------------ |
+| `kind`       | string                            | RECOMMENDED | Describes the type of the resource i.e.`TaskRun` |
+| `apiVersion` | string                            | RECOMMENDED | Schema version i.e. `v1`                         |
+| `metadata`   | [`ObjectMeta`](#objectmeta)       | REQUIRED    | Common metadata about a resource                 |
+| `spec`       | [`TaskRunSpec`](#taskrunspec)     | REQUIRED    | Defines the desired state of TaskRun             |
+| `status`     | [`TaskRunStatus`](#taskrunstatus) | REQUIRED    | Defines the current status of TaskRun            |
 
-For example, in the Kubernetes implementation `[parent]` refers to a Kubernetes namespace. Other implementations could interpret the string differently, including enabling hierarchies of resources containing `TaskRun`s, for example the API endpoint `/apis/tekton.dev/v1beta1/folders/my-folder/project/my-project/taskruns` could represent a hierarchy of "projects" that own `TaskRun`s.
+### `PipelineRun`
 
-| HTTP Verb                 | Requirement      |
-|---------------------------|------------------|
-| Create (POST)             | REQUIRED         |
-| Patch (PATCH)*            | RECOMMENDED      |
-| Replace (PUT)**           | RECOMMENDED      |
-| Delete (DELETE)           | OPTIONAL         |
-| Read (GET)                | REQUIRED         |
-| List (GET)                | REQUIRED         |
-| Watch (GET)               | OPTIONAL         |
-| DeleteCollection (DELETE) | OPTIONAL         |
+A `PipelineRun` represents an instantiation of a single execution of a `Pipeline`. It can describe the spec of the Pipeline directly.
 
-\* Kubernetes only allows JSON merge patch for CRD types. It is recommended that if allowed, at least JSON Merge patch be made available. [JSON Merge Patch Spec (RFC 7386)](https://tools.ietf.org/html/rfc7386)
+| Field        | Type                        | Requirement | Notes                                                 |
+| ------------ | --------------------------- | ----------- | ---------------------------------------------------- |
+| `kind`       | string                      | RECOMMENDED | Describes the type of the resource i.e.`PipelineRun` |
+| `apiVersion` | string                      | RECOMMENDED | Schema version i.e. `v1`                             |
+| `metadata`   | [`ObjectMeta`](#objectmeta) | REQUIRED    | Common metadata about a resource                     |
+| `spec`       | [`PipelineRunSpec`]         | REQUIRED    | Defines the desired state of PipelineRun             |
```

Comment:
nit: looks like we are missing links here

Reply:
thanks!